src/cmake/modules/GUDHI_third_party_libraries.cmake

```
@@ -134,15 +134,17 @@ add_definitions( -DBOOST_SYSTEM_NO_DEPRECATED )
 
 if (WITH_GUDHI_PYTHON)
   # Find the correct Python interpreter.
-  # Can be set with -DPYTHON_EXECUTABLE=/usr/bin/python3 or -DPython_ADDITIONAL_VERSIONS=3 for instance.
-  find_package( PythonInterp )
-  
+  # Can be set with -DPython_EXECUTABLE=/usr/bin/python3 for instance.
+  # CMP0094 OLD finds system python interpreter instead of conda even when using '-DCMAKE_PREFIX_PATH=$CONDA_PREFIX'
+  cmake_policy(SET CMP0094 NEW)
+  find_package( Python COMPONENTS Interpreter)
```

Why not the following, to make it less likely to pick a *bad* version of Python?
```suggestion
  find_package( Python COMPONENTS Interpreter Development.Module NumPy)
```

---

I do not see a lot of added value with Development.Module or NumPy. In the [documentation](https://cmake.org/cmake/help/latest/module/FindPython.html), you can get `Python_NumPy_INCLUDE_DIRS`, that I get in `setup.py.in` with the python code `from numpy import get_include as numpy_get_include`, and `Python_NumPy_VERSION`, that I get with the CMake function `find_python_module` (that performs `${Python_EXECUTABLE}  -c "import ${PYTHON_MODULE_NAME}; print(${PYTHON_MODULE_NAME}.__version__`), and by the way I also need this function for the other python dependencies to decide if a test must be passed or not. Maybe we could also skip tests when `import` throw `ImportError`.

I am not sure this would help to understand this ugly CMakeLists.txt... But tell me if you disagree, I can try

---

> I do not see a lot of added value with Development.Module or NumPy.

I didn't suggest them for the variables they define, but because if they are present, find_package is supposed to find a version of Python that has the dev stuff and numpy installed. If there are several versions of Python installed, this makes it less likely to pick one that is unusable for our purpose. (we do need the dev stuff to compile a plugin)

---

Ok, that makes sense. I added your proposal on [23a393b](https://github.com/GUDHI/gudhi-devel/pull/1112/commits/23a393bf23af0fc7345fff227a615112ec6018ff)